fire_atlas_editor/lib/screens/editor_screen/widgets/rename_atlas_modal.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:fire_atlas_editor/store/actions/atlas_actions.dart';
+import 'package:fire_atlas_editor/store/actions/editor_actions.dart';
+import 'package:fire_atlas_editor/store/store.dart';
+import 'package:fire_atlas_editor/widgets/button.dart';
+import 'package:fire_atlas_editor/widgets/input_text_row.dart';
+import 'package:fire_atlas_editor/widgets/text.dart';
+import 'package:flutter/widgets.dart';
+import 'package:slices/slices.dart';
+
+class RenameAtlasModal extends StatefulWidget {
+  final String? currentName;
+
+  const RenameAtlasModal({Key? key, required this.currentName})
+      : super(key: key);
+
+  @override
+  State createState() => _RenameAtlasModalState();
+}
+
+class _RenameAtlasModalState extends State<RenameAtlasModal> {
+  final newNameController = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
+    newNameController.text = widget.currentName ?? '';
+  }
+
+  @override
+  Widget build(BuildContext ctx) {
+    return Container(
+      padding: const EdgeInsets.only(left: 20, right: 20),
+      child: Column(
+        children: [
+          const FSubtitleTitle(title: 'Rename atlas'),
+          InputTextRow(
+            label: 'New name:',
+            inputController: newNameController,
+          ),
+          const SizedBox(height: 10),
+          Row(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              FButton(
+                label: 'Cancel',
+                onSelect: () => _closeModal(ctx),
+              ),
+              const SizedBox(width: 20),
+              FButton(
+                label: 'Rename',
+                onSelect: () => _renameAtlas(ctx),
+              ),
+            ],
+          ),
+          const SizedBox(height: 20),
+        ],
+      ),
+    );
+  }
+
+  Future<void> _renameAtlas(BuildContext ctx) async {
+    final store = SlicesProvider.of<FireAtlasState>(ctx);
+    final newName = newNameController.text;
+    if (newName.isEmpty) {
+      store.dispatch(
+        CreateMessageAction(
+          type: MessageType.ERROR,
+          message: 'You must inform the name to rename.',
```

Comment:
What does this mean, maybe it can be reformulated slightly? 😄

Reply:
it means you must provide a name in order to rename the project. how would you prefer it to be worded? happy to change :)

Reply:
how about `Name is required` ?

Reply:
```suggestion
          message: 'The new name is required.',
```